sys/log/full/src/log.c

```diff
@@ -320,21 +324,41 @@ log_read_hdr_walk(struct log *log, struct log_offset *log_offset, const void *dp
         }
     }
 
+    if (arg->hdr->ue_flags & LOG_FLAGS_TRAILER_SUPPORT && log->l_th) {
+#if MYNEWT_VAL(LOG_FLAGS_TRAILER_SUPPORT)
+        if (log->l_th->log_process_trailer) {
```

Comment:
This is the only place where `process_trailer` callback is called. How can application actually access the trailer data when reading the log in a generic way?

Reply:
That is exactly why `l_trailer_arg` was there. If an application needs to process the data, they will have to implement the process callback in which the data will get read during the correct function calls.

Reply:
I don't really understand what you are trying to do here.

If you call `log_walk` to read the log there should be a way for the application to get the trailer data for each entry. This can be either by a callback which `log_walk` should call for each entry or (I think preferably) there should be an API that application can use to read that trailer data.